File: frontend/src/services/insumos/Alimentos.ts
// src/services/Alimentos.ts
import API_BASE from "../../config/api";

export type Alimento = {
  id: number;
  tipo: string;
  fase: string;
  proteina: string;
  energiaKcal: string;
  presentacion: string;
  proveedor: string;
  lote: string;
  cantidad: number;
  unidad: string;
  stock: number;
  vencimiento: string;
  observaciones: string;
};

const EMPRESA_ID = 1;
const GRANJA_ID = 1;

function mapApiToAlimento(api: any): Alimento {
  return {
    id: api.id,
    tipo: api.tipo,
    fase: api.fase,
    proteina: api.proteina ?? "",
    energiaKcal: api.energiaKcal ?? api.energia_kcal ?? "",
    presentacion: api.presentacion ?? "",
    proveedor: api.proveedor ?? "",
    lote: api.lote ?? "",
    cantidad: api.cantidad,
    unidad: api.unidad,
    stock: api.stock,
    vencimiento: api.vencimiento ?? "",
    observaciones: api.observaciones ?? "",
  };
}

export async function getAlimentos(): Promise<Alimento[]> {
  const res = await fetch(
    `${API_BASE}/insumos/alimentos/?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`
  );
  if (!res.ok) throw new Error("Error al obtener alimentos");
  const data = await res.json();
  return data.map(mapApiToAlimento);
}

export type NuevoAlimento = Omit<Alimento, "id">;

export async function addAlimento(alimento: NuevoAlimento): Promise<Alimento> {
  const payload = {
    empresa_id: EMPRESA_ID,
    granja_id: GRANJA_ID,
    tipo: alimento.tipo,
    fase: alimento.fase,
    proteina: alimento.proteina,
    energiaKcal: alimento.energiaKcal,
    presentacion: alimento.presentacion,
    proveedor: alimento.proveedor,
    lote: alimento.lote,
    cantidad: alimento.cantidad,
    unidad: alimento.unidad,
    stock: alimento.stock,
    vencimiento: alimento.vencimiento || null,
    observaciones: alimento.observaciones,
  };

  const res = await fetch(`${API_BASE}/insumos/alimentos/`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!res.ok) throw new Error("Error al agregar alimento");
  const data = await res.json();
  return mapApiToAlimento(data);
}

export async function updateAlimento(
  id: number,
  alimento: NuevoAlimento
): Promise<Alimento> {
  const payload = {
    tipo: alimento.tipo,
    fase: alimento.fase,
    proteina: alimento.proteina,
    energiaKcal: alimento.energiaKcal,
    presentacion: alimento.presentacion,
    proveedor: alimento.proveedor,
    lote: alimento.lote,
    cantidad: alimento.cantidad,
    unidad: alimento.unidad,
    stock: alimento.stock,
    vencimiento: alimento.vencimiento || null,
    observaciones: alimento.observaciones,
  };

  const res = await fetch(
    `${API_BASE}/insumos/alimentos/${id}?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`,
    {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload),
    }
  );
  if (!res.ok) throw new Error("Error al actualizar alimento");
  const data = await res.json();
  return mapApiToAlimento(data);
}

export async function deleteAlimento(id: number): Promise<void> {
  const res = await fetch(
    `${API_BASE}/insumos/alimentos/${id}?empresa_id=${EMPRESA_ID}&granja_id=${GRANJA_ID}`,
    { method: "DELETE" }
  );
  if (!res.ok) throw new Error("Error al eliminar alimento");
}

File: backend/services/insumos/Alimentos.py
// src/services/insumosAlimentosService.ts
export type Alimento = {
  id: string;
  tipo: string;
  fase: string;
  proteina: string;
  energiaKcal: string;
  presentacion: string;
  proveedor: string;
  lote: string;
  cantidad: number;
  unidad: string;
  stock: number;
  vencimiento: string;
  observaciones: string;
};

const STORAGE_KEY = "insumos_alimentos";

function getAll(): Alimento[] {
  const raw = localStorage.getItem(STORAGE_KEY);
  return raw ? JSON.parse(raw) : [];
}

function saveAll(lista: Alimento[]) {
  localStorage.setItem(STORAGE_KEY, JSON.stringify(lista));
}

export function getAlimentos() {
  return getAll();
}

export function addAlimento(alimento: Omit<Alimento, "id">) {
  const nuevo: Alimento = { ...alimento, id: crypto.randomUUID() };
  const alimentos = getAll();
  alimentos.push(nuevo);
  saveAll(alimentos);
  return nuevo;
}

export function updateAlimento(id: string, alimento: Omit<Alimento, "id">) {
  const alimentos = getAll();
  const idx = alimentos.findIndex(a => a.id === id);
  if (idx >= 0) {
    alimentos[idx] = { ...alimento, id };
    saveAll(alimentos);
    return alimentos[idx];
  }
  throw new Error("Registro no encontrado");
}

export function deleteAlimento(id: string) {
  const alimentos = getAll().filter(a => a.id !== id);
  saveAll(alimentos);
}
